### PR TITLE
test: restructure integration tests

### PR DESCRIPTION
**Pull request by mattsse**
## Motivation
Following https://matklad.github.io/2021/02/27/delete-cargo-integration-tests.html

## Solution
* move all integration tests into `tests/it`
* feature gate the external integration tests
* run two separate jobs ci jobs, as external integration tests can fail due to changes in the repos

### CODE REVIEW

File: cli/tests/it/cast.rs [hunk 1]
```diff
@@ -24,5 +24,5 @@ casttest!(finds_block, |_: TestProject, mut cmd: TestCommand| {
     println!("{}", output);
 
     // Expect successful block query
-    assert!(output.contains("6574364"));
+    // assert!(output.contains("6574364"), "{}", output);
```
Comment (onbjerg) [inline cli/tests/it/cast.rs:27]: re-enable

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -76,7 +76,28 @@ jobs:
...
+        run: make fmt-testdata
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+      - uses: Swatinem/rust-cache@v1
+        with:
+          cache-on-failure: true
```
Comment (onbjerg) [inline .github/workflows/test.yml:97]: need to add the rpc cache here instead of above

**Review by onbjerg — CHANGES_REQUESTED** [review]

File: cli/tests/it/cast.rs (hunk 1 above)
Comment (mattsse) [reply cli/tests/it/cast.rs:27]: this did fail when ETH_RPC_URL was set -.-

Comment (onbjerg) [reply cli/tests/it/cast.rs:27]: Weird, let's create an issue and fix later

**Review by onbjerg — APPROVED** [review]

File: cli/tests/it/cast.rs (hunk 1 above)
Comment (mattsse) [reply cli/tests/it/cast.rs]: perhaps an issue with the RPC I use locally, let's see

**Pull request merged**